libraries/AP_NavEKF2/AP_NavEKF2.cpp

```diff
@@ -1190,17 +1190,17 @@ uint32_t NavEKF2::getLastYawResetAngle(float &yawAngDelta)
     yaw_reset_data.last_function_call = imuSampleTime_us / 1000;
     yawAngDelta = 0;
     uint32_t lastYawReset_ms = 0;
-    float temp_yawAng;
-    uint32_t lastCoreYawReset_ms = core[primary].getLastYawResetAngle(temp_yawAng);
 
-    // If core has changed (and data not consumed yet) or if the core change was the last yaw reset, return its data
-    if (yaw_reset_data.core_changed || lastCoreYawReset_ms <= yaw_reset_data.last_primary_change) {
+    // There has been a change notification in the primary core that the controller has not consumed
+    if (yaw_reset_data.core_changed) {
```

Comment:
This is wrong. If there was no reset after the core change then it will return an earlier reset event and not the last one (which was the core change).
Can you explain what was the bug you saw here?